purchase_product_packaging_container_deposit/models/purchase_order_line.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class PurchaseOrderLine(models.Model):
+    _name = "purchase.order.line"
+    _inherit = ["purchase.order.line", "container.deposit.order.line.mixin"]
+
+    def _get_product_qty_field(self):
+        return "product_qty"
+
+    def _get_product_qty_delivered_received_field(self):
+        return "qty_received"
+
+    def _compute_qty_received(self):
+        res = super()._compute_qty_received()
+        self.mapped("order_id").update_order_container_deposit_quantity()
```

Comment:
I'm looking again at this and I don't think it's a good idea to trigger other writes in a computed field.
It can be dangerous and trigger side effects you won't expect.
I know that the same code exists in https://github.com/OCA/sale-workflow/pull/2692 but I think I overlooked there.
I'd rather  override `write` or use an inverse method.
In fact, if you override the write in the mixin you can avoid this override here.
You can then check if this field is written, if yes, you call the update.

Reply:
While testing `test_purchase_product_packaging_container_deposit_quantities_case7`
I tried to execute `update_order_container_deposit_quantity` inside `write` or in method `inverse_qty_received`. But this methods does not get triggered when `qty_received` is updated. Therefore I don't see how I can implement your request.

Reply:
I doubt write is not called when you change `qty_received`. We can have a look together.

Reply:
🙏 Thanks let's check together.